internal/cloudproviders/aws/main.go

```diff
@@ -550,3 +550,26 @@ func (obj *AwsProvider) GetRAWClusterInfos(storage types.StorageFactory) ([]clou
 
 	return data, nil
 }
+
+func (obj *AwsProvider) GetKubeconfig(storage types.StorageFactory) (*string, error) {
+
+	_read, err := storage.Read()
+	if err != nil {
+		log.Error("handled error", "catch", err)
+		return nil, err
+	}
+	log.Debug(awsCtx, "data", "read", _read)
+
+	// add logic to fetch the updated kubeconfig
+	// may be store some sort of ttl inside the already existing kubeconfig
+
+	//if _read.ClusterKubeConfig.ttls > time.After()
+	// storage.Write()
+	// InitState(storage, Get) // client initilaiz
+	// obj.client.GetKubeconfig()
+	//}
```

Comment:
So whenever the user invokes ksctl switch it will trigger this and each time a new session token is created which will be used. if this is the case there might be a `MAX_LIMIT_SESSION` which can have problems if we are triggereing it without checking if the session already there is exeiting and if its there we can delete from the aws side and a new token from there